packages/front-end/pages/getstarted/data-source-guide.tsx

```diff
@@ -0,0 +1,271 @@
+import { PiCheckCircleFill } from "react-icons/pi";
+import { useState, useEffect } from "react";
+import Link from "next/link";
+import { getDemoDatasourceProjectIdForOrganization } from "shared/demo-datasource";
+import { Box, Grid, Separator } from "@radix-ui/themes";
+import DocumentationSidebar from "@/components/GetStarted/DocumentationSidebar";
+import UpgradeModal from "@/components/Settings/UpgradeModal";
+import { useUser } from "@/services/UserContext";
+import PageHead from "@/components/Layout/PageHead";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import LoadingOverlay from "@/components/LoadingOverlay";
+import { useGetStarted } from "@/services/GetStartedProvider";
+import Frame from "@/ui/Frame";
+import DataSourceDiagram from "@/components/InitialSetup/DataSourceDiagram";
+import ViewSampleDataButton from "@/components/GetStarted/ViewSampleDataButton";
+
+// Also used for the `Launch Setup Flow` button to keep it aligned
+const DOCUMENTATION_SIDEBAR_WIDTH = "minmax(0, 245px)";
+
+const DataSourceGuide = (): React.ReactElement => {
+  const { organization } = useUser();
+  const [upgradeModal, setUpgradeModal] = useState<boolean>(false);
+  const {
+    factTables,
+    datasources,
+    ready: definitionsReady,
+    project,
+    factMetrics,
+    error,
+  } = useDefinitions();
+  const { setStep, clearStep } = useGetStarted();
+
+  const demoProjectId = getDemoDatasourceProjectIdForOrganization(
+    organization.id || "",
+  );
+
+  // If they view the guide, clear the current step
+  useEffect(() => {
+    clearStep();
+  }, [clearStep]);
+
+  const loading = !definitionsReady;
+
+  if (loading) {
+    return <LoadingOverlay />;
+  }
+
+  if (error) {
+    return <div className="alert alert-danger">{error}</div>;
+  }
+
+  // Ignore the demo datasource for all checks
+  const hasFactTables = project
+    ? factTables.some(
+        (f) =>
+          f.projects.includes(project) && !f.projects.includes(demoProjectId),
+      )
+    : factTables.some((f) => !f.projects.includes(demoProjectId));
+  const hasFactMetrics = project
+    ? factMetrics.some(
+        (m) =>
+          m.projects.includes(project) && !m.projects.includes(demoProjectId),
+      )
+    : factMetrics.some((m) => !m.projects.includes(demoProjectId));
+  const hasDatasource = project
+    ? datasources.some(
+        (d) =>
+          d.projects?.includes(project) && !d.projects?.includes(demoProjectId),
+      )
+    : datasources.some((d) => !d.projects?.includes(demoProjectId));
```

Comment:
Fact tables, metrics, and data sources can also be global (empty projects array). Those are being skipped here when they shouldn't be